server/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -746,6 +757,16 @@ public boolean hasClusterObjects() {
             return clusterInfo.keySet().size() > 0;
         }
 
+        /**
+         * @return true if this Clusters object has been initialized with remote Cluster objects
+         *              This will be false for local-cluster (non-CCS) only searches.
+         */
+        public boolean hasRemoteClusters() {
+            if (clusterInfo.size() > 1) {
+                return true;
+            }
+            return clusterInfo.size() > 0 && clusterInfo.containsKey(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY) == false;
```

Comment:
Maybe use `clusterInfo.size() == 1` here instead? Also maybe cache the size to a variable, makes this easier to read.

Reply:
There already is a `total` field. So I simplified the logic to:

```
            return total > 1 || clusterInfo.keySet().stream().anyMatch(alias -> alias != RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY);

```